app/controllers/events_controller.rb

```diff
@@ -0,0 +1,79 @@
+class EventsController < ApplicationController
+  helper EventsHelper
+  before_action :authenticated?
+  before_action :set_event, only: %i[show edit update destroy]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @future_events = Event.upcoming
+    @past_events = Event.past
+    @my_past = Event.my_past(current_user[:id])
+    @my_upcoming = Event.my_upcoming(current_user[:id])
```

Comment:
- [ ] try using current_user.my_events.past

Reply:
we tried that but it doesn't work